include/net/nrf_cloud.rst

```diff
@@ -43,15 +43,18 @@ When the poll thread option is used directly with the nRF Cloud library, the enu
 First, the library tries to establish the transport for communicating with the cloud.
 This procedure involves a TLS handshake that might take up to three seconds.
 The API blocks for the duration of the handshake.
-
-Next, the API subscribes to an MQTT topic to start receiving user association requests from the cloud.
 The cloud uses the certificates of the device for authentication.
 See `Updating the nRF Connect for Cloud certificate`_ and the :ref:`modem_key_mgmt` library for more information on modem credentials.
 The certificates are generated using the device ID and PIN/HWID.
+The default format for the device ID is ``nrf-<IMEI>``.  This format is valid only for Nordic hardware, such as a Thingy:91 or nRF9160 DK.
+For custom hardware, use a prefix other than ``nrf-`` by modifying :option:`CONFIG_NRF_CLOUD_CLIENT_ID_PREFIX`.
+To use a UUID as the device ID, set :c:define:`NRF_CLOUD_CLIENT_ID` to the desired UUID string.
+
+Next, the API subscribes to an MQTT topic to start receiving user association requests from the cloud.
 
 Every time nRF Connect for Cloud starts a communication session with a device, it verifies whether the device is uniquely associated with a user.
 If not, the user association procedure is triggered.
-When adding the device to an nRF Connect for Cloud account, the user must provide the correct device ID and PIN (for Thingy:91) or HWID (for nRF9160 DK) to nRF Cloud.
+When adding the device to an nRF Connect for Cloud account, the user must provide the correct device ID and PIN (for Thingy:91 and custom hardware) or HWID (for nRF9160 DK) to nRF Cloud.
```

Comment:
This statement is a bit misleading. I suggest: "To add a device to an nRF Connect for Cloud account you have three options: 1) the [ProvisionDevices endpoint](https://api.nrfcloud.com/v1#operation/ProvisionDevices); 2) the nRFCloud.com Web app; 3) the [AssociateDevice endpoint](https://api.nrfcloud.com/v1#operation/AssociateDevice). For options 2 and 3 you need to provide the PIN (Thingy:91 or custom hardware) or HWID (nRF9160 DK) that is embedded in your device's certificate. This is visible on the label for the Thingy:91 and nRF9160 DK if you are using the device certificate that was shipped with it. Otherwise, for custom hardware or certificates obtained from the [CreateDeviceCertificate endpoint](https://api.nrfcloud.com/v1#operation/CreateDeviceCertificate), you need to enter the same PIN that you entered when creating the certificate."

Reply:
perhaps it would be better to remove this and just point to a single guide: "adding/provisioning nrf cloud devices".
i'm not sure if this readme is the right place for referencing specific API endpoints.
@umapraseeda what do you suggest?

Reply:
@jayteemo I can see your point. A guide could be linked to from other places, too, so it gets more use. We have an open Jira ticket for a guide to connect non-Nordic products to nRF Cloud, but we could expand that to cover Nordic products, too.

Reply:
I agree @jayteemo. We should link to this information once it is in the guide.